mythril/laser/ethereum/svm.py

```diff
@@ -106,35 +98,44 @@ def __init__(
 
         log.info("LASER EVM initialized with dynamic loader: " + str(dynamic_loader))
 
-    @property
-    def accounts(self) -> Dict[str, Account]:
-        """
-
-        :return:
-        """
-        return self.world_state.accounts
-
     def sym_exec(
-        self, main_address=None, creation_code=None, contract_name=None
+        self,
+        world_state: WorldState = None,
+        target_address: int = None,
+        creation_code: str = None,
+        contract_name: str = None,
     ) -> None:
+        """ Starts symbolic execution
+        There are two modes of execution.
+        Either we analyze a preconfigured configuration, in which case the world_state and target_address variables
+        must be supplied.
+        Or we execute the creation code of a contract, in which case the creation code and desired name of that
+        contract should be provided.
+
+        :param world_state The world state configuration from which to perform analysis
+        :param target_address The address of the contract account in the world state which analysis should target
+        :param creation_code The creation code to create the target contract in the symbolic environment
+        :param contract_name The name that the created account should be associated with
         """
+        pre_configuration_mode = world_state is not None and target_address is not None
+        scratch_mode = creation_code is not None and contract_name is not None
+        if pre_configuration_mode == scratch_mode:
+            raise ValueError("Symbolic execution started with invalid parameters")
 
-        :param main_address:
-        :param creation_code:
-        :param contract_name:
-        """
         log.debug("Starting LASER execution")
         for hook in self._start_sym_exec_hooks:
             hook()
 
         time_handler.start_execution(self.execution_timeout)
         self.time = datetime.now()
 
-        if main_address:
-            log.info("Starting message call transaction to {}".format(main_address))
-            self._execute_transactions(main_address)
+        if pre_configuration_mode:
+            self.open_states = [world_state]
+            log.info("Starting message call transaction to {}".format(target_address))
+            self._execute_transactions(symbol_factory.BitVecVal(target_address, 256))
 
-        elif creation_code:
+        elif scratch_mode:
+            # self.open_states = [WorldState()]
```

Comment:
any reason why this is commented?

Reply:
Yeah, execute contract creation set's up the global state

Reply:
I mean this can be removed right?

Reply:
it is 😅